src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs

```diff
@@ -12,15 +12,8 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
 {
     internal static class TracingUtils
     {
-        static private HostType? _hostType = null;
-
         public static HostType GetHostType()
         {
-            if (_hostType.HasValue)
```

Comment:
There is some value in having the `_hostType` field cache the result. We don't swallow exceptions on every call from lines 54-59. Can we go with a solution where we tie the _hostType evaluation to the lifetime of the configuration provider that is using it?

Reply:
Yes, that's the current behavior. We retrieve the host type information once and then cache it at the provider level.

Reply:
Okay, sounds good.